### DISCUSSION TITLE
iPXE does not process "root-path" option.

### DISCUSSION THREAD

**Discussion by tokohito** [ipxe/ipxe#220] [Q&A]
Hello

I challenging to boot FreeBSD installer from iPXE(UEFI).
FreeBSD UEFI bootloader(loader.efi) needs "root-path" dhcp option(option num is 17).
I tried to set "root-path" at dnsmasq and/or ipxe scripts.
However, neither approach not be able to pass the parameters correctly to FreeBSD installer.

- dnsmasq.conf
```
dhcp-option=17,"nfs://172.16.0.13/freebsd-12.2/"
dhcp-option=encap:175,8,1
dhcp-option=encap:175,9,1
```

- ipxe
```
:freebsd12
#set root-path nfs://172.16.0.10/freebsd-12.2/
set root-path 172.16.0.10:/freebsd-12.2/
echo root-path....... 017.... ${root-path}   <--- works even without the above set command
chain tftp://172.16.0.10/freebsd-12.2/boot/loader.efi
boot
```

FreeBSD installer called from ipxe stop at the following message step.
"freebsd loader.efi setting currdev to net0"

Please give me some advice to make install FreeBSD successful with ipxe.
best regards.

**Comment by mcb30** [comment]
From observing packet captures, it looks as though FreeBSD will perform its own DHCP transaction, rather than using the parameters as captured and provided by iPXE.

Various threads (e.g. https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=202098) suggest that you ought to be able to use the `dhcp.root-path=...` syntax to override the DHCP-provided root path.  For example:

```
:freebsd12
chain tftp://172.16.0.10/freebsd-12.2/boot/loader.efi dhcp.root-path=172.16.0.10:/freebsd-12.2/
```

**Reply by tokohito** [reply]
Thank you for your advice.
I tried again with the advice, but unfortunately it was the same as the previous case...
(By the way, I'm trying both dhcp.root-path and dhcp.root-path.override option)

If loader.efi called directly from Onboard PXE ROM without using iPXE, loader.efi will interpret the root-path dhcp option value and proceed, so I suspect that there is a problem with iPXE.
However, this is very strange, but even when loader.efi called from Onboard PXE ROM, NFS did not work properly and could only be load by TFTP.
(This NFS Server can be used without problems by connecting from other Linux.)

**Reply by vestein463** [reply]
My experience with booting FreeBSD via ixpe is that proxy dhcp is not possible. FreeBSD insists on getting root-path from the  same server as the other IP info. DId you use proxy dhcp?
